addons/xivbar/xivbar.lua

```diff
@@ -0,0 +1,216 @@
+              documentation and/or other materials provided with the distribution.
+            * Neither the name of xivbar nor the
+              names of its contributors may be used to endorse or promote products
+              derived from this software without specific prior written permission.
+
+        THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+        ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+        WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+        DISCLAIMED. IN NO EVENT SHALL SirEdeonX BE LIABLE FOR ANY
+        DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+        (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+        LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+        ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+        (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+        SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+
+-- Addon description
+_addon.name = 'XIVBar'
+_addon.author = 'Edeon'
+_addon.version = '1.0'
+_addon.language = 'english'
+
+-- Libs
+config = require('config')
+texts  = require('texts')
+images = require('images')
+
+-- User settings
+local defaults = require('defaults')
+local settings = config.load(defaults)
+config.save(settings)
+
+-- Load theme options according to settings
+local theme = require('theme')
+local theme_options = theme.apply(settings)
+
+-- Addon Dependencies
+local ui = require('ui')
+local player = require('player')
+local xivbar = require('variables')
+
+-- initialize addon
+function initialize()
+    ui:load(theme_options)
+
+    local windower_player = windower.ffxi.get_player()
+
+    if windower_player ~= nil then
+        player.hpp = windower_player.vitals.hpp
+        player.mpp = windower_player.vitals.mpp
+        player.current_hp = windower_player.vitals.hp
+        player.current_mp = windower_player.vitals.mp
+        player.current_tp = windower_player.vitals.tp
+
+        player:calculate_tpp()
+    end
+
+    xivbar.initialized = true
+end
+
+-- update a bar
+function update_bar(bar, text, width, current, pp, flag)
+    local old_width = width
+    local new_width = math.floor((pp / 100) * theme_options.bar_width)
+
+    if new_width ~= nil and new_width >= 0 then
+        if old_width == new_width then
+            if new_width == 0 then
+                bar:hide()
+            end
+
+            if flag == 1 then
+                xivbar.hp_update = false
+            elseif flag == 2 then
+                xivbar.update_mp = false
+            elseif flag == 3 then
+                xivbar.update_tp = false
+            end
+        else
+            local x = old_width
+
+            if old_width < new_width then
+                x = old_width + math.ceil(((new_width - old_width) * 0.1))
```

Comment:
There are extra parentheses around this and the lower expression... yeah, that's the best kind of criticism I can come up with now :D